pkg/kv/kvserver/replica_raft.go

```diff
@@ -795,7 +795,7 @@ func (r *Replica) handleRaftReadyRaftMuLocked(
 		if hasReady = raftGroup.HasReady(); hasReady {
 			// Since we are holding raftMu, only this Ready() call will use
 			// raftMu.bytesAccount. It tracks memory usage that this Ready incurs.
-			r.raftMu.bytesAccountUse = true
+			r.raftMu.bytesAccountUse = RaftEntriesMemLimit > 0
```

Comment:
Perhaps we need to enable this unconditionally, to always track memory usage, even if it's not limited.